python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/asset_condition_tests/test_result_value_hash.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+from dagster import (
+    AssetSpec,
+    DailyPartitionsDefinition,
+    # doing this rename to make the test cases fit on a single line for readability
+    SchedulingCondition as SC,
+)
+
+from dagster_tests.definitions_tests.auto_materialize_tests.base_scenario import run_request
+
+from ..scenario_specs import ScenarioSpec
+from .asset_condition_scenario import SchedulingConditionScenarioState
+
+one_parent = ScenarioSpec(asset_specs=[AssetSpec("A"), AssetSpec("downstream", deps=["A"])])
+two_parents = ScenarioSpec(
+    asset_specs=[AssetSpec("A"), AssetSpec("B"), AssetSpec("downstream", deps=["A", "B"])]
+)
+
+daily_partitions = DailyPartitionsDefinition(start_date="2020-01-01")
+one_parent_daily = one_parent.with_asset_properties(partitions_def=daily_partitions)
+two_parents_daily = two_parents.with_asset_properties(partitions_def=daily_partitions)
+
+
+@pytest.mark.parametrize(
+    ["expected_value_hash", "condition", "scenario_spec", "materialize_A"],
+    [
+        # cron condition returns a unique value hash if parents change, if schedule changes, if the
+        # partitions def changes, or if an asset is materialized
+        ("a92dbabace17f6e01d1401b24fa8ef91", SC.on_cron("0 * * * *"), one_parent, False),
```

Comment:
@alangenfeld here are those promised tests -- it's slightly different than you might have imagined from the previous PR because I moved the value hash generation logic off of the serialized AssetConditionEvaluation object onto the in-memory SchedulingResult (because that object doesn't actually need a value hash, it's just the cursor that needs it)

So this test just generates a result in a stable way and makes sure we get a stable value hash. This way, if we accidentally introduce a change that changes how we generate value hashes, we can be aware of that. In most cases, it's fine if we change hashes across the board from release to release (the cost is just that the first tick after upgrading will write a new evaluation record for all assets, even if nothing really changed), what we really want to catch are cases where we introduce a change that causes the value hash to be completely unstable (i.e. you get a completely unique hash every time you run this test)